docs/tutorial/object_decomp.md

```diff
@@ -1,14 +1,161 @@
+
+### Extracting assets
+
+You can run `extract_assets.py` to extract the object's assets. Running it with no arguments extracts *everything* all over again, though. A better way is to run it with `-s` (for single file), and give it the location of the object you want to extract relative to `assets`, i.e.
+
+```bash
+./extract_assets.py -s objects/object_name
+```
+
+This should populate the folder you created earlier. ZAPD produces a C file containing the extracted object data, which will be `assets/objects/object_name/object_name.c`. Any data that you have not specified the type of, or is not referenced elsewhere in the object, is extracted as unknown blobs  (usually named `unaccounted_XXXXXX`).  Open the C file to see if there are any such blobs. (Some are just padding with 0s and can be ignored.)
+
+You now have to try and decipher these blobs using the [list of tools given below](#tools) to work out what they might be. In the case of unused parts of the object, this can be very difficult.
+
+### Naming
+
+You'll want to name everything that exists in the XML, but it helps to break it down into separate steps. After each step, extract the single asset again.
+
+1. First, name every piece of data that is directly accessed by another part of the repo (generally, but not always, actors). This allows for elimination of entries in [`undefined_syms.txt`](../../undefined_syms.txt). For most objects, this includes several of a skeleton, animations, some textures, and collision information.
+
+2. Then, name the limbs associated with each skeleton. This is usually straightforward, providing Z64Utils or Hylian Toolbox can show the skeleton properly. Note that when naming limbs, you should name them *from the object's perspective*, so the "left arm" is the one that is to the *object's* left. As part of this step, you should also replace the autogenerated enum values associated with each limb with an appropriate name.
+
+3. Next, name the display lists that are associated to limbs in the skeleton. Z64Utils can show you which display list is associated with a given limb.
+
+4. If any non-limb display lists still lack a name, try naming them now.
+
+5. Lastly, try to name every texture in the object. This can be a bit tricky just by looking at the textures, so it helps to see how they're used in the display lists. You can also try manipulating the textures, recompiling the game, and loading up the resulting ROM to see what your changes look like in-game.
+
+You'll notice that we didn't mention vertices in this list. This is because, for the most part, vertices don't really appear in the XML, and you don't need to name them. However, some actors have code that manipulates or otherwise interacts with vertices; in these cases, the vertices should be added to the XML and named.
```

Comment:
```md
don't really appear in the XML
```

I think we can be a bit more precise here,
```suggestion
You'll notice that we didn't mention vertices in this list. This is because, for the most part, vertices are not used outside the object itself, and so are not required to have names. However, some actors have code that manipulates or otherwise interacts with vertices; in these cases, the vertices should be added to the XML and named.
```